iphone/Classes/TiAppiOSProxy.m

```diff
@@ -507,15 +495,50 @@ - (NSArray *)supportedUserActivityTypes
 
 - (NSDictionary *)currentUserNotificationSettings
 {
-  __block NSDictionary *returnVal = nil;
-  TiThreadPerformOnMainThread(^{
-    UIUserNotificationSettings *notificationSettings = [[UIApplication sharedApplication] currentUserNotificationSettings];
-    returnVal = [[self formatUserNotificationSettings:notificationSettings] retain];
-  },
-      YES);
+  DEPRECATED_REPLACED(@"App.iOS.currentUserNotificationSettings", @"7.2.0", @"App.iOS.NotificationCenter.requestUserNotificationSettings");
 
-  return [returnVal autorelease];
-  ;
+  if ([TiUtils isIOS10OrGreater]) {
+    DebugLog(@"[ERROR] Please use Ti.App.iOS.NotificationCenter.requestUserNotificationSettings in iOS 10 and later to request user notification settings asynchronously.");
+    return;
```

Comment:
If this function is previously getting used, now it will not return relevant data in iOS >= 10.

Reply:
Updated!